rosbag2_tests/test/rosbag2_tests/test_rosbag2_play_end_to_end.cpp

```diff
@@ -158,12 +160,12 @@ TEST_F(PlayEndToEndTestFixture, play_filters_by_topic) {
   EXPECT_THAT(exit_code, Eq(EXIT_SUCCESS));
 
   EXPECT_THAT(primitive_messages, SizeIs(Ge(0u)));
-  EXPECT_THAT(array_messages, SizeIs(Ge(2u)));
+  EXPECT_THAT(array_messages, SizeIs(Ge(1u)));
 
   // Play all topics
   sub_ = std::make_unique<SubscriptionManager>();
-  sub_->add_subscription<test_msgs::msg::BasicTypes>("/test_topic", 3);
-  sub_->add_subscription<test_msgs::msg::Arrays>("/array_topic", 2);
+  sub_->add_subscription<test_msgs::msg::BasicTypes>("/test_topic", 2);
```

Comment:
One thing that might be nice in here is to define these numbers as named constant variables at the top of the test case, so that it's more obvious which ones should match, what's being checked

Reply:
e.g.

```
const uint num_messages_to_wait_for  = 2;
...

add_subscription<>("", num_messages_to_wait_for);

...

EXPECT_THAT(primitive_messages, SizeIs(Ge(num_messages_to_wait_for));

Reply:
f78c2e3